Add tests

GerryFudd opened the pull request:
This PR includes unit tests for the existing API and mocks for rsa keys and database data.

GerryFudd committed:
setup basic test structure

GerryFudd committed:
added more database mocks

GerryFudd committed:
don't lose these

GerryFudd committed:
Merge branch 'master' into add-tests

GerryFudd committed:
moved tests to their new home

GerryFudd committed:
Merge branch 'master' into add-tests

GerryFudd committed:
don't keep my personal vscode settings checked into version control

GerryFudd committed:
don't keep my personal settings checked into version control

GerryFudd committed:
moved test files and started setting up unit tests to run with new dockerizes config

GerryFudd committed:
tests work

GerryFudd committed:
updated test setup and added tests that use auth

GerryFudd committed:
finish testing mocks and notes

GerryFudd committed:
Merge branch 'master' into add-tests

grantal reviewed (approved):
Looks good!

grantal reviewed (changes requested):
You should update the requirements.txt to include pytest

marty-hope commented:
In order to execute the tests through travis, please add pytest to the requirements.txt file.

ickybyte committed:
Add pytest v4.2.1 to requirements.txt
This will allow the travis CI config to install
pytest and allow the unit tests to be executed.

ickybyte committed:
Update travis.yml
Add line that installs package dependencies defined in requirements.txt.

ickybyte committed:
Fix path to be relative

ickybyte committed:
FIx pip command.
--user option reported as an error by travis.
The command "pip install --user -r ./WEB/requirements.txt"

marty-hope committed:
Merge pull request #15 from CodeForPortland/add-package-dependencies

Add package dependencies

ickybyte committed:
Execute unit tests
Add commands to travis.yml to execute tests

ickybyte committed:
Specify python version

grantal committed:
add requirements instruction

grantal committed:
move unit tests to the script section
